src/Adafruit_SPIFlash.cpp

```diff
@@ -31,6 +31,12 @@ enum
   EXTERNAL_FLASH_DEVICE_COUNT = sizeof(possible_devices)/sizeof(possible_devices[0])
 };
 
+Adafruit_SPIFlash::Adafruit_SPIFlash()
+  : _cache()
+{
+  _trans = NULL;
```

Comment:
since _trans can be `NULL` now, we should have a  check for it in begin() function, something like
```
bool Adafruit_SPIFlash::begin(void)
{
  if (!_trans) return false;
  ..........
}
```